src/spac/spatial_analysis.py

```diff
@@ -120,25 +159,75 @@ def Cluster_Interaction_Matrix_Analysis(
         fig = plt.figure()
         ax = fig.add_subplot(1, 1, 1)
 
+    # Operational Section
+    # -----------------------------------------------
+
     # Create a categorical column data for plotting
     new_annotation_name = annotation + "_plot"
 
     adata.obs[new_annotation_name] = pd.Categorical(
         adata.obs[annotation])
 
+    if stratify_by:
+        if isinstance(stratify_by, list):
+            adata.obs[
+                'concatenated_obs'
+            ] = adata.obs[stratify_by].astype(str).agg('_'.join, axis=1)
+        else:
+            adata.obs[
+                'concatenated_obs'
+            ] = adata.obs[stratify_by]
+
     # Compute a connectivity matrix from spatial coordinates
-    sq.gr.spatial_neighbors(adata)
+    if stratify_by:
+        ax_dictionary = {}
+        unique_values = adata.obs['concatenated_obs'].unique()
+        buffer = io.BytesIO()
+        pickle.dump(ax, buffer)
+        for subset_key in unique_values:
+            # Subset the original AnnData object based on the unique value
+            subset_adata = adata[
+                adata.obs[
+                    'concatenated_obs'
+                ] == subset_key
+            ].copy()
 
-    if analysis_method == "Neighborhood Enrichment":
-        ax = Neighborhood_Enrichment_Analysis(
-                adata,
-                new_annotation_name,
-                ax)
+            sq.gr.spatial_neighbors(subset_adata)

```

Comment:
@ruiheesi, this part of the code and if statements at line 201 run again at line 218. Is this block of code a good candidate for  intermediate level function?

Reply:
Sounds good, added.